packages/astro/src/runtime/client/dev-overlay/plugins/xray.ts

```diff
@@ -41,6 +41,10 @@ export default {
 				const tooltip = buildIslandTooltip(island);
 				attachTooltipToHighlight(highlight, tooltip, islandElement);
 
+				// Set the z-index to be 1 higher than the greatest z-index in the stack.
+				const zIndex = getHighlightZIndex(islandElement);
+				tooltip.style.zIndex = highlight.style.zIndex = zIndex+'';
```

Comment:
Just noting that setting this to some high number like `9999` would work as well. I thought it would be better to allow other things to possibly go on top, like other plugins. So instead this calculates the next `z-index` for an element.